WordPress/openverse-frontend
# More TS-ification and refactoring

## Fixes


No issue closed but this belongs to the [TypeScriptification milestone](https://github.com/WordPress/openverse-frontend/milestone/10).

## Description



- src
	- **assets/licenses** | renamed to match CC's [press kit](https://mirrors.creativecommons.org/presskit/icons/) & license elements
	- components
 		- License
     	 	- **VLicense.vue** | updated to use the `getElements` util added to `utils/license.ts`
      		- **VLicenseElements.vue** | updated to use the `getElements` util added to `utils/license.ts`
			- meta
       	 		- **VLicenseElements.stories.mdx** | added stories, test in Storybook
    	- VContentReport
    		- **VContentReportForm.vue** | updated to match flat constants structure
    		- **VReportDescForm.vue** | updated to match flat constants structure
    	- VErrorSection
    	    - **VErrorSection.vue** | function renamed, no change
		- VFilters
			- **VLicenseExplanation.vue** | import path change, can ignore
		- **VLink.vue** | minor type change, can ignore
  		- VMediaInfo
  			- **VCopyLicense.vue** | hollow out component using new attribution util `getAttribution` 😎 
  			- **VMediaLicense.vue** | import path change, can ignore
  			- **VMediaReuse.vue** | removed unused props passed to component that doesn't accept them
  	- constants
		- **license.ts** | added constants and types for license elements
	- models
		- **media.ts** | replaced `string` with types `License` and `LicenseVersion`
	- pages/audio
		- **_id.vue** | removed unused props passed to component that doesn't accept them
	- utils
		- **attribution-html.ts** | _converged all markup generation logic into one function_ 💪
		- **license.ts** | wrote a util to split a license into it's constituent elements
- tests/unit/specs/utils | added/updated tests for the above changes

The main code change to review in this PR is `attribution-html.ts`. I tried my best to keep it modular and readable but I'm happy to provide a walkthrough, perform further refactors if the code is too messy.

## Testing Instructions

Since this is a refactor PR, play around with with the site (touching upon the pages with the affected components) and ensure everything works like before. Ideally nothing should be broken.

Also check out the Storybook for some added/fixed stories.

## Checklist


[best_practices]:https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin


<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>
